superset-frontend/src/SqlLab/components/RunQueryActionButton.tsx

```diff
@@ -17,53 +17,57 @@
  * under the License.
  */
 import React from 'react';
-import PropTypes from 'prop-types';
 import { t } from '@superset-ui/translation';
 
 import Button from '../../components/Button';
 
-const propTypes = {
-  allowAsync: PropTypes.bool.isRequired,
-  dbId: PropTypes.number,
-  queryState: PropTypes.string,
-  runQuery: PropTypes.func.isRequired,
-  selectedText: PropTypes.string,
-  stopQuery: PropTypes.func.isRequired,
-  sql: PropTypes.string.isRequired,
-};
-const defaultProps = {
-  allowAsync: false,
-  sql: '',
-};
+interface Props {
+  allowAsync: boolean;
+  dbId: number | null;
+  queryState: string;
```

Comment:
can we make this `queryState?: string` and remove the default queryState?

Reply:
Didn't change this one since queryState is being using in _indexOf_ few lines below which gets type-angry if you feed *undefined* to it.